example/rails_app/lib/event_handler.rb

```diff
@@ -2,6 +2,6 @@ module EventHandler
   extend self
 
   def handle_event event
-    puts "RailsApp::EventHandler - Consumed event: #{event.name} with payload: #{event.payload}"
+    puts "RailsApp::EventHandler - Consumed event: #{event.metadata[:name]} with payload: #{event.payload}"
```

Comment:
Ha! I agree that that looked cleaner. I see the benefit of splitting out the payload and metadata, but I see less benefit for splitting the name out of the metadata. Would helper methods to directly access fields from the metadata be beneficial?